controllers/elfmachine_controller_placement_group.go

```diff
@@ -453,28 +459,57 @@ func (r *ElfMachineReconciler) migrateVMForPlacementGroup(ctx *context.MachineCo
 		return false, err
 	}
 
-	if *kcp.Spec.Replicas != kcp.Status.UpdatedReplicas {
+	// KCP is in rolling update.
+	if *kcp.Spec.Replicas != kcp.Status.Replicas {
+		cpElfMachines, err := machineutil.GetControlPlaneElfMachinesInCluster(ctx, ctx.Client, ctx.Cluster.Namespace, ctx.Cluster.Name)
+		if err != nil {
+			return false, err
+		}
+
+		var newCPElfMachinesInPG []*infrav1.ElfMachine
+		for i := 0; i < len(cpElfMachines); i++ {
+			if ctx.ElfMachine.Name != cpElfMachines[i].Name &&
+				cpElfMachines[i].Status.PlacementGroupRef == *placementGroup.ID &&
+				cpElfMachines[i].CreationTimestamp.After(ctx.ElfMachine.CreationTimestamp.Time) {
+				newCPElfMachinesInPG = append(newCPElfMachinesInPG, cpElfMachines[i])
+			}
+		}
+
+		if len(newCPElfMachinesInPG) == int(*kcp.Spec.Replicas-1) {
```

Comment:
478行之下加个comment: 
// This is the last CP ElfMachine which has not been added into the target PlacementGroup.
// Migrate this VM to the target host, then it will be added into the target PlacementGroup.